File: hrms-frontend/src/services/userService.js
import axios from "axios"

export default class UserService {

    existsByEmail(email){
        return axios.get(`http://localhost:8080/api/users/exists/byEmail?email=${email}`)
    }

    existsByEmailAndPW(email, password){
        return axios.get(`http://localhost:8080/api/users/exists/byEmailAndPW?email=${email}&password=${password}`)
    }

    deleteById(userId) {
        return axios.delete(`http://localhost:8080/api/users/delete/byId?userId=${userId}`)
    }

    login(email, password) {
        return axios.get(`http://localhost:8080/api/users/login?email=${email}&password=${password}`)
    }

    updateEmail(userId, email){
        return axios.put(`http://localhost:8080/api/users/update/email?email=${email}&userId=${userId}`)
    }

    updatePassword(userId, password, oldPassword){
        return axios.put(`http://localhost:8080/api/users/update/pw?oldPassword=${oldPassword}&password=${password}&userId=${userId}`)
    }

    updateProfileImg(imgId, userId) {
        return axios.put(`http://localhost:8080/api/users/update/profileImg?${imgId ? `imgId=${imgId}&` : ""}userId=${userId}`)
    }

}

File: hrms-backend/src/main/java/com/finalproject/hrmsbackend/core/business/abstracts/UserService.java
package com.finalproject.hrmsbackend.core.business.abstracts;

import com.finalproject.hrmsbackend.core.utilities.results.DataResult;
import com.finalproject.hrmsbackend.core.utilities.results.Result;

public interface UserService {

    DataResult<Boolean> existsByEmail(String email);

    DataResult<Boolean> existsByEmailAndPW(String email, String password);

    Result deleteById(int userId);

    Result login(String email, String password);

    Result updateEmail(String email, int userId);

    Result updatePW(String password, String oldPassword, int userId);

    Result updateProfileImg(Integer imgId, int userId);
}

